spec/suites/map/MapSpec.js

```diff
@@ -333,6 +475,112 @@ describe("Map", function () {
 		});
 	});
 
+	describe("setMinZoom and #setMaxZoom", function () {
+		describe("when map is not loaded", function () {
+			it("change min and max zoom but not zoom", function () {
+				map.setZoom(2);
+				map.setMinZoom(3);
+
+				expect(map.getZoom()).to.eql(2);
+				expect(map.getMinZoom()).to.eql(3);
+
+				map.setMaxZoom(7);
+
+				expect(map.getZoom()).to.eql(2);
+				expect(map.getMaxZoom()).to.eql(7);
+			});
+
+			it("do not fire 'zoomlevelschange'", function () {
+				var spy = sinon.spy();
+				map.on("zoomlevelschange", spy);
+
+				map.setZoom(5);
+				map.setMinZoom(3);
+				map.setMaxZoom(7);
+
+				expect(map.getZoom()).to.eql(5);
+				expect(map.getMinZoom()).to.eql(3);
+				expect(map.getMaxZoom()).to.eql(7);
+
+				expect(spy.called).to.not.be.ok();
+			});
+		});
+
+		describe("when map is loaded", function () {
+			var spy;
+
+			beforeEach(function () {
+				map.setView([0, 0], 4); // loads map
+
+				spy = sinon.spy();
+				map.on("zoomlevelschange", spy);
+			});
+
+			it("do not fire 'zoomlevelschange' if zoom level did not change", function () {
+				map.setMinZoom(2);
+				map.setMaxZoom(7);
+
+				expect(map.getZoom()).to.eql(4);
+				expect(map.getMinZoom()).to.eql(2);
+				expect(map.getMaxZoom()).to.eql(7);
+				expect(spy.calledTwice).to.be.ok();
+
+				var postSpy = sinon.spy();
+				map.on("zoomlevelschange", postSpy);
+
+				map.setMinZoom(2);
+				map.setMaxZoom(7);
+
+				expect(postSpy.called).to.not.be.ok();
+			});
+
+			it("fire 'zoomlevelschange' but do not change zoom if max/min zoom is less/more current zoom", function () {
+				map.setMinZoom(2);
+				map.setMaxZoom(7);
+
+				expect(map.getZoom()).to.eql(4);
+				expect(map.getMinZoom()).to.eql(2);
+				expect(map.getMaxZoom()).to.eql(7);
+				expect(spy.calledTwice).to.be.ok();
+			});
+
+			// unable to meet conditions to change zoom; can't change zoom if map is loaded due to setView
+			it.skip("fire 'zoomlevelschange' and change zoom if max/min zoom is more/less current zoom", function () {
+				map.setMinZoom(5);
+
+				expect(map.getZoom()).to.eql(5);
+				expect(map.getMinZoom()).to.eql(5);
+				expect(spy.called).to.be.ok();
+
+				map.setZoom(8);
+				map.setMaxZoom(7);
+
+				expect(map.getZoom()).to.eql(7);
+				expect(map.getMaxZoom()).to.eql(7);
+				expect(spy.calledTwice).to.be.ok();
+			});
```

Comment:
Not sure if the intended function is to move to the appropriate zoom or actual change the zoom value.

```
if (this.getZoom() < this.options.minZoom) {
	return this.setZoom(zoom);
}
```